Refactor and modularize user guide (#826)
pydata/pydata-sphinx-theme: docs/user_guide/light-dark.rst

.. _manage-themes:

Light and dark themes
=====================

You can change the major background / foreground colors of this theme according to "dark" and "light" modes.
These are controlled by a button in the navigation header, with the following options:

- A ``light`` theme with a bright background and dark text / UI elements
- A ``dark`` theme with a dark background and light text / UI elements
- ``auto``: the documentation theme will follow the system default that you have set

Configure default theme mode
----------------------------

By default, visitors to your documentation will use the theme mode ``auto``.

> This will choose a theme based on the user's system settings, and default to ``light`` if not settings are present.

Reviewer: ```suggestion
This will choose a theme based on the user's system settings, and default to ``light`` if no system setting is found.
```

If you wish to use a different default theme mode, set the ``default_mode`` configuration to one of ``auto``, ``dark``, ``light``.
For example:

.. code-block:: python

   html_context = {
      # ...
      "default_mode": "light"
   }

For more information, see :ref:`manage-themes`.

.. tip::

   To completely remove the theme management, configure ``default_mode`` to the value you want in your documentation (``light`` or ``dark``) and then remove the theme-switcher from the ``navbar_end`` section of the header navbar configuration:

   .. code-block:: python

      html_theme_options {

>           # ...

Reviewer: ```suggestion
          ...
```
for consistency (elsewhere we don't comment out the `...`, even though it's technically not valid Python)

          # Note we have omitted `theme-switcher` below
          "navbar_end": ["navbar-icon-links"]
      }

Customize the CSS of light and dark themes
------------------------------------------

.. danger::

>     Theming is still a beta feature so the variables related to the theme switch are likely to change in the future. No backward compatibily is guaranteed when customization is done.

Reviewer: ```suggestion
    Theming is still a beta feature so the variables related to color theming are likely to change in the future. No backward compatibily is guaranteed when customization is done.
```

To customize the CSS of page elements in a theme-dependent manner, use the ``html[data-theme='<THEME>']`` CSS selector.
For example to define a different background color for both the light and dark themes:

.. code-block:: css

    /* anything related to the light theme */
    html[data-theme="light"] {

        /* whatever you want to change */
        background: white;
    }

    /* anything related to the dark theme */
    html[data-theme="dark"] {

        /* whatever you want to change */
        background: black;
    }

> A complete list of the used colors for this theme can be found in the :doc:`CSS style section <styling>`.

Reviewer: ```suggestion
A complete list of the colors used in this theme can be found in the :doc:`CSS style section <styling>`.
```
